hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java

```diff
@@ -997,13 +997,17 @@ protected RequestFactory createRequestFactory() {
     String storageClassConf = getConf()
         .getTrimmed(STORAGE_CLASS, "")
         .toUpperCase(Locale.US);
-    StorageClass storageClass;
-    try {
-      storageClass = StorageClass.fromValue(storageClassConf);
-    } catch (IllegalArgumentException e) {
-      LOG.warn("Unknown storage class property {}: {}; falling back to default storage class",
-          STORAGE_CLASS, storageClassConf);
-      storageClass = null;
+    StorageClass storageClass = null;
+    if (!storageClassConf.isEmpty()) {
+      try {
+        storageClass = StorageClass.fromValue(storageClassConf);
+      } catch (IllegalArgumentException e) {
+        LOG.warn("Unknown storage class property {}: {}; falling back to default storage class",
+            STORAGE_CLASS, storageClassConf);
+      }
+    } else {
+      LOG.info("Empty storage class property {}; falling back to default storage class",
```

Comment:
We print configuration related logs in debug only unless they are supposed to cause error. So I think this should be debug. right @steveloughran

Reply:
Also shouldn't the log message say " Unset" rather than empty?

Reply:
yes, debug only, apart from some special cases where there's a risk of incompatibility across versions (dir marker policy). even there i think it is time to retire it.